cumulus/client/relay-chain-minimal-node/src/lib.rs

```diff
@@ -127,6 +128,7 @@ pub async fn build_minimal_relay_chain_node_with_rpc(
 	let client = cumulus_relay_chain_rpc_interface::create_client_and_start_worker(
 		relay_chain_url,
 		task_manager,
+		polkadot_config.prometheus_registry(),
```

Comment:
I am thinking about whether this is better registered on the parachain side.

Technically, this is doing relay chain calls. However, the calls to the relay chain are basically done only on collators, the code lives in cumulus. As a user I would probably expect these metrics to be attached to the parachain prometheus endpoint.

Reply:
I looked for reasons to keep the metrics on the relay chain side but couldn't find any. To be honest, it is still not clear to me what kind of metrics should fall under the relay chain prometheus exporter (its concerns on the collator side are not crispy clear in my mind yet), but for our case I agree that these metrics seem more relevant to the internals of how parachains work, so it would be useful to expose them in the "parachain" prometheus exporter exporter.

Changed this here: [309ae23](https://github.com/paritytech/polkadot-sdk/pull/5572/commits/309ae2390d438417a264beef27933db0e5ee86c6).

Reply:
> To be honest, it is still not clear to me what kind of metrics should fall under the relay chain prometheus exporter

So when you don't use `--relay-chain-rpc-urls` then the collator will start an embedded node. Which is basically the same as a polkadot full node that you start with the `polkadot` binary. 

If course this internal node will export all its metrics, there is a whole bunch of them defined in substrate. So in these scenarios it can make sense to monitor the relay chain node separately. This RPC functionality however is in the end parachain specific and therefore goes to the parachain prometheus endpoint.